examples/remix/app/routes/conditional-ssr.tsx

```diff
@@ -0,0 +1,46 @@
+import { json, LoaderFunction } from '@remix-run/node';
+import { useLoaderData } from '@remix-run/react';
+import { getSupabase, User } from '@supabase/auth-helpers-remix';
+import { Database } from '../../db_types';
+
+// this route demonstrates how to subscribe to fetch the user
+// on the server to ensure it is available for first SSR render
+export const loader: LoaderFunction = async ({
+  request
+}: {
+  request: Request;
+}) => {
+  const response = new Response();
+  const supabaseClient = getSupabase<Database>({ request, response });
+
+  const {
+    data: { session }
+  } = await supabaseClient.auth.getSession();
```

Comment:
can simplify:

```js
const {
    data: { user }
  } = await supabaseClient.auth.getUser();
```

Reply:
It's probably best to keep it as `getSession`, since this function will refresh the session if necessary or get the session from inMemory while `getUser` won't do this.

Reply:
Good call!